backend/modules/simulation/services/progress_service.py

```
@@ -0,0 +1,122 @@
+        user_progress = self.repository.get_user_progress_by_id(user_progress_id)
+        if not user_progress:
+            raise NotFoundError("User progress not found")
+        
+        if user_progress.user_id != user_id:
+            raise ForbiddenError("Access denied")
+        
+        simulation = self.repository.get_simulation_by_id(user_progress.simulation_id)
+        if not simulation:
+            raise NotFoundError("Simulation not found")
+        
+        scenes = self.repository.get_scenes_by_simulation_id(user_progress.simulation_id)
```

Ruff (0.14.8)

Local variable `scenes` is assigned to but never used

Remove assignment to unused variable `scenes`

(F841)